lib/active_record/connection_adapters/sqlserver_adapter.rb

```diff
@@ -119,6 +119,8 @@ def config_timeout(config)
         def config_encoding(config)
           config[:encoding].present? ? config[:encoding] : nil
         end
+
+        def configure_application_name; end
```

Comment:
Changing this means the README is wrong. I think we should remove this method altogether and move the configuration to `database.yml`.

There are 2 problems:

1. Before removing, we should deprecate its usage so users have time to fix their applications
2. The current behaviour is incorrect and it is failing silently

What I think should be done:

1. Fix `config_appname` so it warns users that their config is wrong if they overwrote `configure_application_name` and suggest them to use `appname` in their `database.yml`
2. Change docs to suggest using `database.yml` instead